### ISSUE TITLE
Wire up class data loader by interface.

### ISSUE THREAD

**Issue by caoquendo** [ChilliCream/hotchocolate#926]
**Describe the bug**
I have implemented a class DataLoaders. It extends the DataLoaderBase class and implements a custom interface by extending the IDataLoader interface, as indicated in https://greendonut.io/docs/custom-dataloader.

When I inject the dataloader to one of my resolver's methods by referencing to the concrete implementation class and using the [DataLoader] attribute, all works fine. 

When I try to inject the dataloader that has an interface (as described in https://hotchocolate.io/docs/resolvers#dataloader) I get an exception that says "The specified class must have exactly one public constructor". My data loader class definitely has a public constructor, so I think the ActivatorHelper component might be trying to instantiate the interface instead.

I might be missing some configuration for this data loaders with interfaces to work, but I think I have applied all the relevant details from the documentation.

Thanks for the help!

**Environment**
 - ASP.NET Core 2.2
 - HotChocolate 9.0.4

**Comment by michaelstaib** [comment]
I will put this one in preview 41

**Comment by michaelstaib** [comment]
@caoquendo

The DataLoader could already resolve dependencies from the dependency injection container. With this change it is now possible to register a DataLoader with the standard dependency injection container. This enables referencing DataLoaders through interfaces. 

Here is how we can now register a DataLoader:

```charp
services.AddDataLoader<IMyDataLoader, MyDataLoader>();
```

```charp
services.AddDataLoader<MyDataLoader>();
```

```charp
services.AddDataLoader<IMyDataLoader>(s => ....);
```

The DataLoaderRegistry is automatically registered when using this.

On the resolver side I can now resolve my DataLoader through an interface:

```charp
public async Task<string> ResolveSomething(IMyDataLoader dataLoader) 
{

}
```

We also do not need to use the `[DataLoader]` attribute anymore if the interface implements `IDataLoader`.

Fixes #966
